### PR TITLE
Minor fixes, main change is in ResavePopup.java, needed to not have Netbeans throw a fit

### PR DESCRIPTION
**Pull request by nicost**

### CODE REVIEW

File: src/main/java/mpicbg/spim/postprocessing/deconvolution2/LRFFT.java [hunk 1]
```diff
@@ -390,8 +390,8 @@ public void setKernel( final Image<FloatType> kernel )
 	public Image<FloatType> getKernel1() { return kernel1; }
 	public Image<FloatType> getKernel2() { return kernel2; }
 	
-	public void setCurrentIteration( final int i ) { this.i = i; }
-	public int getCurrentIteration() { return i; }
+	public void setCurrentIteration( final int i ) { this.iteration = i; }
```
Comment (dscho) [inline src/main/java/mpicbg/spim/postprocessing/deconvolution2/LRFFT.java]: The `this.` prefix now becomes unnecessary. BTW I really like the more descriptive name `iteration`.

File: src/main/java/mpicbg/spim/postprocessing/deconvolution2/LRFFT.java [hunk 2]
```diff
@@ -441,7 +441,7 @@ public void setKernel( final Image<FloatType> kernel )
 				final FourierConvolution<FloatType, FloatType> fftConv = fftConvolution1;
 				fftConv.replaceImage( image );
 				fftConv.process();
-				System.out.println( " block " + i + ": compute " + (System.currentTimeMillis() - time) );
+				System.out.println(" block " + iteration + ": compute " + (System.currentTimeMillis() - time) );
```
Comment (dscho) [inline src/main/java/mpicbg/spim/postprocessing/deconvolution2/LRFFT.java]: I fear that @StephanPreibisch's coding style disagrees with the rest of Fiji's, instead opting for some style concocted in the ImgLib2 project that is a variation of *some* C++ projects' coding style (and unfortunately, it is not used consistently, as can be seen in the hunk above where function bodies consisting of one line are on the same line as the function declaration, which violates that coding style).

The proper solution would be to document the coding style of *this* Fiji plugin (as it is an outlier when it comes to Fiji's source code), or to switch to [Fiji's source code conventions](http://fiji.sc/Coding_style).

Until that happens, I fear that the closest indication as to @StephanPreibisch's wishes is the surrounding code.

File: src/main/java/spim/fiji/plugin/Image_Fusion.java [hunk 3]
```diff
@@ -35,13 +35,13 @@
 
 public class Image_Fusion implements PlugIn
 {
-	public static ArrayList< Fusion > staticFusionAlgorithms = new ArrayList< Fusion >();
+	public final static ArrayList< Fusion > staticFusionAlgorithms = new ArrayList< Fusion >();
```
Comment (dscho) [inline src/main/java/spim/fiji/plugin/Image_Fusion.java]: This is a backwards-incompatible change that would require the major version number to increment, as per @StephanPreibisch's pledge (unfortunately hidden to the public in a private email conversation) to heed [SemVer](http://semver.org): `Image_Fusion` is a public class, therefore part of SPIM_Registration's public API, and before this change it is possible to compile this code (while compilation will fail after this change):

```java
// Find that bugger that removed my algorithm
Image_Fusion.staticFusionAlgorithms = new ArrayList< Fusion >( Image_Fusion.staticFusionAlgorithms ) {
    @Override
    public Fusion remove( int i ) {
        throw new RuntimeException( "I do not want anything to be removed!" );
    }
};
```

Note that this is not a theoretical problem, as such code constructs are often needed when highly skilled software developers debug hard problems.

File: src/main/java/spim/fiji/spimdata/explorer/popup/ResavePopup.java [hunk 4]
```diff
@@ -57,7 +59,7 @@ public ResavePopup()
 	}
 
 	@Override
-	public JMenuItem setViewExplorer( final ViewSetupExplorerPanel< ?, ? > panel )
+	public JMenuItem setViewExplorer(ViewSetupExplorerPanel<? extends AbstractSpimData<? extends AbstractSequenceDescription<?, ?, ?>>, ?> panel )
```
Comment (dscho) [inline src/main/java/spim/fiji/spimdata/explorer/popup/ResavePopup.java]: This is also a backwards-incompatible change.

File: src/main/java/spim/fiji/spimdata/explorer/popup/ResavePopup.java [hunk 5]
```diff
@@ -160,7 +159,7 @@ else if ( index == 2 || index == 3 ) // HDF5, compressed HDF5
 						final int firstviewSetupId = data.getSequenceDescription().getViewSetupsOrdered().get( 0 ).getId();
 						final ExportMipmapInfo autoMipmapSettings = perSetupExportMipmapInfo.get( firstviewSetupId );
 
-						final boolean compress = ( index == 2 ) ? false : true;
+						final boolean compress = (index != 2);
```
Comment (dscho) [inline src/main/java/spim/fiji/spimdata/explorer/popup/ResavePopup.java]: The parentheses are unnecessary (and were so even before this change).

**Comment by StephanPreibisch** [comment]
@nicost @dscho  I just merged it, looks all ok so far ... Just a heads up, there is another major change coming up soon anyways, so all the ImgLib1 deconvolution code where you made some changes will be gone.

**Comment by dscho** [comment]
Oh well. I knew I shouldn't have spent the effort.

**Comment by StephanPreibisch** [comment]
Thanks a lot for going over it and implicitly reminding me that I missed that one @dscho ... I increased the pom.xml version, so this is all fine. Sorry for being a bit short on comments, but I just have 2 minute breaks from experiments and microscopy for the next 2-3 weeks. Any help is always greatly appreciated!

File: src/main/java/spim/fiji/plugin/Image_Fusion.java (hunk 3 above)
Comment (nicost) [inline src/main/java/spim/fiji/plugin/Image_Fusion.java]: Aj, my bad!  I have to admit that the Java convention of considering everything to be a public api unless declared as "internal" to be very confusing (it makes much more sense to me to consider everything "internal" unless exposed as an external api).  If anyone else uses this code, please do revert my change (and remove final).

**Pull request merged**